src/java/org/apache/cassandra/guardrails/Guardrails.java

```diff
@@ -41,203 +36,214 @@
  */
 public abstract class Guardrails
 {
+    public static final String CUSTOM_GUARDRAILS_FACTORY_PROPERTY = "cassandra.custom_guardrails_factory_class";
+
     private static final GuardrailsConfig config = DatabaseDescriptor.getGuardrailsConfig();
 
+    public static final GuardrailsFactory factory = System.getProperty(CUSTOM_GUARDRAILS_FACTORY_PROPERTY) == null
```

Comment:
As it was pointed out to me, I will also pass along the recommendation to use `CassandraRelevantProperties` here rather than `System.getProperty`.